### PR TITLE
OP-338 - Powershell startup script

### PR DESCRIPTION
**Pull request by mizzioisf**
See https://openhospital.atlassian.net/browse/OP-338.

Release of the new windows powershell startup script oh.ps1.
Updated oh.bat and oh.sh to align scripts.

### CODE REVIEW

File: poh-bundle-win/oh.bat [hunk 1]
```diff
@@ -23,7 +23,7 @@ REM #
 REM ################### Configuration ###################
```
Comment (pviotti) [inline poh-bundle-win/oh.bat:23]: wasn't this file meant to be replaced by the poweshell script - or I'm missing something..?

File: poh-bundle-win/oh.ps1 [hunk 2]
```diff
@@ -0,0 +1,844 @@
...
+	switch -Regex ( $MYSQL_DIR ) {
+		"mariadb" {
+	    	Start-Process -FilePath "$OH_PATH\$MYSQL_DIR\bin\mysql_install_db.exe" -ArgumentList ("--datadir=$OH_PATH\$DATA_DIR --password=$MYSQL_ROOT_PW") -Wait -NoNewWindow -RedirectStandardOutput "$LOG_DIR/$LOG_FILE" -RedirectStandardError "$LOG_DIR/$LOG_FILE_ERR"
+        	}
+		"mysql" {
+		    Start-Process "$OH_PATH\$MYSQL_DIR\bin\mysqld.exe" -ArgumentList ("--initialize-insecure --basedir=$OH_PATH\$MYSQL_DIR --datadir=$OH_PATH\$DATA_DIR") -Wait -NoNewWindow -RedirectStandardOutput "$LOG_DIR/$LOG_FILE" -RedirectStandardError "$LOG_DIR/$LOG_FILE_ERR"; break
+        }
+	}
+
+#	if ( $? -ne 0 ){
```
Comment (pviotti) [inline poh-bundle-win/oh.ps1:411]: btw what about all the commented out lines? Are those meant for devops/admin to tweak the script or can they be removed?

**Review by pviotti — COMMENTED** [review]
looks good to me (and I agree with you on PowerShell... it does make sense in some contexts, but for simple scripts like this one it's certainly not as concise and elegant as Bash).
Haven't tried it to run it though (not sure I'll have time to do it anytime soon, sorry).

File: poh-bundle-linux-x32/oh.sh [hunk 3]
```diff
@@ -144,24 +142,34 @@ SCRIPT_NAME=$(basename "$0")
...
+        echo "   -l    set language: en|fr|it|es|pt"
+        echo "   -C    start OH - CLIENT mode (Client / Server configuration)"
+        echo "   -t    test database connection (CLIENT mode only)"
+        echo "   -D    start OH in DEMO mode"
+        echo "   -d    start OH in DEBUG mode"
+        echo "   -c    clean OH installation"
+        echo "   -G    setup GSM"
+        echo "   -v    show OH software version and configuration"
+        echo "   -h    show this help"
+        echo "   -q    quit"
```
Comment (mwithi) [inline poh-bundle-linux-x32/oh.sh]: Question, should we put in alphabetical order like in man?

File: poh-bundle-win/oh.bat (hunk 1 above)
Comment (mizzioisf) [reply poh-bundle-win/oh.bat:23]: Yes. We are still not really sure that the powershell script is ok for every scenario, we can remove the oh.bat after some good testing I think.

File: poh-bundle-win/oh.ps1 (hunk 2 above)
Comment (mizzioisf) [reply poh-bundle-win/oh.ps1:411]: The script still does not manage error exit code in the mysql functions. I left it there (bash way :-) ) so if anyone want's to give it a try...
Unfortunately the start-process function doesn't seem to handle this...

File: poh-bundle-linux-x32/oh.sh (hunk 3 above)
Comment (mizzioisf) [reply poh-bundle-linux-x32/oh.sh]: This is in alphabetical order (linux version):

```
 ---------------------------------------------------------
|                                                         |
|                   Open Hospital | OH                    |
|                                                         |
 ---------------------------------------------------------
 lang en | arch x86_64

 Usage: oh.sh.1 [ -lang en|fr|it|es|pt ] 

   -c    clean OH installation
   -C    start OH in CLIENT mode (Client / Server configuration)
   -d    start OH in DEBUG mode
   -D    start OH in DEMO mode
   -G    setup GSM
   -h    show this help
   -l    set language: en|fr|it|es|pt
   -r    restore OH database
   -s    save OH database
   -t    test database connection (CLIENT mode only)
   -v    show OH software version and configuration

```
What do you think ? Some options might be confused...

Comment (mwithi) [reply poh-bundle-linux-x32/oh.sh]: it is very clear to me

Comment (mwithi) [reply poh-bundle-linux-x32/oh.sh]: seen also the new option -X for the clean, ok for me!

File: poh-bundle-win/oh.bat (hunk 1 above)
Comment (mwithi) [reply poh-bundle-win/oh.bat:23]: I think the new version of `oh.bat` solves everything, until `oh.ps1` alone will not work OOTB

File: poh-bundle-win/oh.ps1 (hunk 2 above)
Comment (mwithi) [reply poh-bundle-win/oh.ps1:411]: Let's merge to close the Sprint, improvements are welcome with a new PR!

**Pull request merged**